network/src/validator_network/health_checker.rs

```diff
@@ -68,28 +68,16 @@ impl HealthCheckerNetworkSender {
     ///
     /// The rpc request can be canceled at any point by dropping the returned
     /// future.
-    pub async fn ping(
+    pub async fn send_rpc(
         &mut self,
         recipient: PeerId,
-        req_msg: Ping,
+        req_msg: HealthCheckerMsg,
         timeout: Duration,
-    ) -> Result<Pong, RpcError> {
+    ) -> Result<HealthCheckerMsg, RpcError> {
```

Comment:
The interface is speaking proto and so I tried to minimize the lines of code changed in the PR and figured this would be handled in the follow up that completely stripped out proto. Happy to do a little more in this PR though.

Reply:
Actually if we do this, we'd end up with some spaghetti  code as the healther checker owns ping but its sending interface uses it and it uses its sending interface. My recommendation is to rethink how we want these validator network pieces to live and ideally in another pr.

Reply:
Ahh just hit this problem as well over in discovery land. Looks like the right path forward is probably extracting all of the `validator_network/*.rs` interfaces into their owning crate, which should remove the weird cyclic dependency issue. Feel free to leave this unchanged in this diff then.